src/VecchiaMLE_kernels.jl

```diff
@@ -24,7 +51,37 @@ function vecchia_mul!(y::Vector{T}, B::Vector{Matrix{T}}, x::Vector{T}, n::Int,
     return y
 end
 
-function vecchia_build_B!(B::Vector{Matrix{T}}, samples, rowsL, colptrL, hess_obj_vals, n::Int, m::Vector{Int}) where T <: AbstractFloat
+@kernel function vecchia_build_B_kernel!(hess_obj_vals, @Const(samples), @Const(rowsL), @Const(colptrL),
+                                         @Const(m), @Const(n), @Const(r))
+    index = @index(Global)
+    pos = colptrL[index]
+    mj = m[index]
+
+    k = 0
+    for s in 1:mj
+        for t in s:mj
+            acc = 0.0
+            for i = 1:r
```

Comment:
Can't one move this inner loop to outer loop?

Reply:
We can't because it's a dot product between `samples[s]` and `samples[t]`.